.github/actions/build-backend/action.yml

```diff
@@ -46,9 +76,121 @@ runs:
+    - name: Spin up containers
+      if: inputs.run-integration-tests == 'true'
+      working-directory: prime-router
+      id: spin_containers
+      run: docker-compose -f docker-compose.yml up -d prime_dev sftp azurite
+      shell: bash
+
+    - name: Give some time for the container to start
+      if: inputs.run-integration-tests == 'true'
+      working-directory: prime-router
+      run: |
+        echo "sleeping 30"
+        sleep 30
+        echo "wake up"
+      shell: bash
+
+    - name: Load the configuration
+      if: inputs.run-integration-tests == 'true'
+      working-directory: prime-router
+      run: |
+        ./gradlew reloadTables
+        ./gradlew reloadSettings
+      shell: bash
+
+    # Integrations tests require the metadata catalog
+    - name: Run Integration Tests
+      if: inputs.run-integration-tests == 'true'
+      working-directory: prime-router
+      run: ./gradlew testIntegration -Pshowtests
+      shell: bash
+
+    - name: Publish Integration Test Results
+      uses: EnricoMi/publish-unit-test-result-action@283dea176069279a9076e77b548668a8e4f0c31b
+      # Per https://github.com/EnricoMi/publish-unit-test-result-action#support-fork-repositories-and-dependabot-branches
+      if: >
+        always() &&
+        github.event.sender.login != 'dependabot[bot]' &&
+        ( github.event_name != 'pull_request' || github.event.pull_request.head.repo.full_name == github.repository ) &&
+        inputs.run-integration-tests == 'true'
+      with:
+        # This path is from the root of the repo as needed by the plugin
+        files: prime-router/build/test-results/testIntegration/**/*.xml
+        check_name: "Integration Test Results"
+
+    - name: Correct sftp perms
+      if: inputs.run-integration-tests == 'true'
+      working-directory: prime-router
+      run: docker-compose exec -T sftp chmod 777 /home/foo/upload
+      shell: bash
+
+    - name: Smoke tests
+      if: inputs.run-integration-tests == 'true'
+      working-directory: prime-router
+      run: |
+        export $(xargs < .vault/env/.env.local)
+        ./prime create-credential --type=UserPass --persist=DEFAULT-SFTP --user foo --pass pass
+        ./gradlew testSmoke
+      shell: bash
+
+    - name: Dump docker logs
+      if: always() && inputs.run-integration-tests == 'true'
+      uses: jwalton/gh-docker-logs@59c9656cd3cb7542525f3dce7ae2f44c0ff85d66
+      with:
+        images: "prime-router_prime_dev,vault"
+
+    - name: Cleanup Gradle Cache
+      if: inputs.run-integration-tests == 'true'
+      working-directory: prime-router
+      run: |
+        rm -f .gradle/caches/modules-2/modules-2.lock
+        rm -f .gradle/caches/modules-2/gc.properties
+      shell: bash
+
     - name: Upload Artifact
       uses: actions/upload-artifact@a8a3f3ad30e3422c9c7b888a15615d19a852ae32
       with:
         name: prime-router-build-${{ inputs.version }}
         path: prime-router/prime-router-build.tar.gz
         retention-days: 3
+
```

Comment:
These steps were previously a separate job.  They are likely to add 10 min. to the full build time.